### PR TITLE
Beacon

### PR DESCRIPTION
**Pull request by yunqingyi**
## Merge checklist
- [ ] Docker image or CCS Project builds with the changes
- [ ] Docker image or CCS Project runs with the changes
- [ ] Can send commands from a mocked ground station/relevant changes tested
- [ ] Changes have been reviewed

### CODE REVIEW

File: Services/include/housekeeping/housekeeping_service.h
Comment (Taylorr82) [inline Services/include/housekeeping/housekeeping_service.h:34]: The project properties should be modified to include this include directory rather than a path name like this

File: Services/include/housekeeping/housekeeping_service.h
Comment (Taylorr82) [inline Services/include/housekeeping/housekeeping_service.h:48]: Was this reformatted by clang?

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_to_beacon.c:23]: again, modify the project include path rather than use paths

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_to_beacon.c:52]: why use a loop instead of one memcpy?

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_to_beacon.c:65]: Are you able to use #defines instead of magic numbers?

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_to_beacon.c:84]: Complete this todo. use int\<n\>_t to specify the length of the integer

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_to_beacon.c:117]: Could you do this

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: Services/include/housekeeping/housekeeping_service.h
Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_service.h:48]: possibly. this changed after the stash popped, but it looks fine so I kept it

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:52]: you're right, I was following what was done before. I'll change all the for loops to one memcpy

File: Services/include/housekeeping/housekeeping_service.h
Comment (Taylorr82) [reply Services/include/housekeeping/housekeeping_service.h:48]: Do you have the pre-commit hooks installed? It looks like clang's formatting got messed up. This should be formatted according to clang

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:65]: the enum defined right before this function should change these magic numbers into something more readable right

File: Services/include/housekeeping/housekeeping_service.h
Comment (arrooney) [reply Services/include/housekeeping/housekeeping_service.h:48]: yeah something is messed with clang format somewhere -- it's changed all the tabs to 2 spaces

File: Services/include/housekeeping/housekeeping_service.h
Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_service.h:34]: done

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:65]: I don't know what these magic numbers are for to tell you

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:23]: done

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:84]: if you look at the next line, I think it needs to be fixed at the driver, and the driver will be changed later because we will be using a different chip. I've created a clickup task for this.

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:117]: I can't lol, looking through the ADCS manual it is unclear what the max angular rate will be. Hence I'm not sure how to change it. The ADCS team will investigate. Another clickup task for later

**Comment by Taylorr82** [comment]
Please resolve the clang-format issues before approval will be granted

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:117]: Make it an issue

Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:117]: we have: TODOs in the code
corresponding to: clickup tasks

do you want an issue for triple redundancy or are we okay with the above

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:65]: these numbers came from the EPS ICD, they are predefined, hence line 65 is changing them from their predefined value to "Power_on" and so on

File: Services/include/housekeeping/housekeeping_service.h
Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_service.h:48]: will the formatting stop the code from working? or are we just discussing a style preference

Comment (arrooney) [reply Services/include/housekeeping/housekeeping_service.h:48]: No it'll run - but it's a detriment to code quality to have inconsistent style. Also in this case it's distracting from the work you've done since a lot of non-changes are being shown (i.e. changes in tab widths) which makes it hard to see what was actually changed, and hence hard to review.

Comment (arrooney) [reply Services/include/housekeeping/housekeeping_service.h:48]: We should review the formatting workflow at the work session - perhaps having it done automatically is not what we want.

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:65]: In this case would something like
switch (all_hk_data->EPS_startup_hk.last_reset_reason_reg):
case (POWER_ON): // do stuff break;
case (...)

be better? This sort of conversion seems a bit strange

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:117]: We'll discuss clickup vs issues at the work session. I'm not sure what the best way forward is

File: Services/include/housekeeping/housekeeping_service.h
Comment (Taylorr82) [reply Services/include/housekeeping/housekeeping_service.h:48]: Also, merging this PR with the formatting changed the way it is will cause a lot of conflicts for anyone else with active branches in this repository. We need to maintain a consistent style and I think automatic tools are the easiest solution

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:23]: Did you change this line?

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_to_beacon.c:84]: thank you

File: Services/source/housekeeping/housekeeping_to_beacon.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_to_beacon.c:65]: I'll write it like this then:

EPS_reset_TypeDef EPS_last_reset_reason;
  switch (all_hk_data->EPS_startup_hk.last_reset_reason_reg) {
    case 0x0C800000:
    EPS_last_reset_reason = Power_on;
    break;
    case 0x24000000:
    EPS_last_reset_reason = IWDG;
    break;
    case 0x04000000:
    EPS_last_reset_reason = NRST;
    break;
    case 0x14000000:
    EPS_last_reset_reason = Software;
    break;
  }

File: Services/include/housekeeping/housekeeping_service.h
Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_service.h:48]: I'll install clang, but for now the format should be manually fixed in the new commit

File: Services/source/housekeeping/housekeeping_service.c
Comment (Taylorr82) [inline Services/source/housekeeping/housekeeping_service.c:993]: Looks like this whole file was converted to two space indentation. It should be 4

**Review by Taylorr82 — CHANGES_REQUESTED** [review]
Looks like all of the files still have some tab width issues. They should be using 4 spaces

File: Services/include/housekeeping/housekeeping_service.h
Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_service.h]: File has been formatted with clang

File: Services/source/housekeeping/housekeeping_service.c
Comment (yunqingyi) [reply Services/source/housekeeping/housekeeping_service.c:996]: file has been reformatted with clang!

**Review by nsorense4 — APPROVED** [review]
I think all of the changes look good. It looks like everything is in clang format now. We will likely need to merge this with all of the changes I made to the housekeeping types.

**Review by Taylorr82 — APPROVED** [review]

File: Services/source/housekeeping/housekeeping_service.c
Comment (nsorense4) [inline Services/source/housekeeping/housekeeping_service.c:234]: This should be NUM_TEMP_SENSOR, not 2.

Comment (nsorense4) [reply Services/source/housekeeping/housekeeping_service.c:234]: Unless we don't want to use that macro because it's in another repo?

Comment (Taylorr82) [reply Services/source/housekeeping/housekeeping_service.c:234]: It would be a good idea to use it

**Review by Taylorr82 — APPROVED** [review]

File: Services/include/housekeeping/housekeeping_to_beacon.h
Comment (Taylorr82) [inline Services/include/housekeeping/housekeeping_to_beacon.h:40]: beacon_task.h doesn't exist, did you mean beacon.h?

Comment (yunqingyi) [reply Services/include/housekeeping/housekeeping_to_beacon.h:40]: beacon.h has been changed to beacon_task.h

**Pull request closed without merging**
